containers-backpack.cabal

```diff
@@ -0,0 +1,189 @@
+cabal-version:       3.0
+name:                containers-backpack
+version:             0.0.0.0
+synopsis:            Backpack signatures for Map-like interfaces
+description:         Backpack signatures for Map-like interfaces
+homepage:            https://github.com/kowainik/containers-backpack
+bug-reports:         https://github.com/kowainik/containers-backpack/issues
+license:             MPL-2.0
+license-file:        LICENSE
+author:              Dmitrii Kovanikov, Veronika Romashkina
+maintainer:          Kowainik <xrom.xkov@gmail.com>
+copyright:           2018-2019 Kowainik
+category:            Backpack, Data, Containers
+build-type:          Simple
+extra-doc-files:     CHANGELOG.md
+                     README.md
+tested-with:         GHC == 8.2.2
+                     GHC == 8.4.4
+                     GHC == 8.6.5
+                     GHC == 8.8.1
+
+source-repository head
+  type:                git
+  location:            https://github.com/kowainik/containers-backpack.git
+
+common common-options
+  build-depends:       base >= 4.10.1.0 && < 4.14
+
+  ghc-options:         -Wall
+                       -Wincomplete-uni-patterns
+                       -Wincomplete-record-updates
+                       -Wcompat
+                       -Widentities
+                       -Wredundant-constraints
+                       -Wmissing-export-lists
+                       -Wpartial-fields
+                       -fhide-source-paths
+  if impl(ghc >= 8.8.1)
+    ghc-options:       -Wmissing-deriving-strategies
+                       -Werror=missing-deriving-strategies
+
+  default-language:    Haskell2010
+  default-extensions:  ConstraintKinds
+                       DeriveGeneric
+                       DerivingStrategies
+                       InstanceSigs
+                       GeneralizedNewtypeDeriving
+                       LambdaCase
+                       OverloadedStrings
+                       RecordWildCards
+                       ScopedTypeVariables
+                       StandaloneDeriving
+                       TupleSections
+                       TypeApplications
+                       ViewPatterns
+
+-- signatures for containers
+library sig
```

Comment:
So there is no difference between internal libraries and libraries to export?

Reply:
It's my fault, turns out that all these libraries are indeed internal at the moment. To make them public, I needed to add `visibility: public` field. And, again, the documentation doesn't really help here...

https://www.haskell.org/cabal/users-guide/developing-packages.html#pkg-field-library-visibility
https://www.haskell.org/cabal/users-guide/developing-packages.html#pkg-section-library-library

Reply:
:disappointed: we need to write more about such things in public so other people at least would be able to find these tricks easier...